vendor/github.com/avinetworks/sdk/go/session/avisession.go

```diff
@@ -546,13 +546,17 @@ func (avisess *AviSession) restRequest(verb string, uri string, payload interfac
 		avisess.collectCookiesFromResp(resp)
 
 		if resp.StatusCode == 401 && len(avisess.sessionid) != 0 && uri != "login" {
-			resp.Body.Close()
 			glog.Infof("Retrying url %s; retry %d due to Status Code %d", url, retry, resp.StatusCode)
 			err := avisess.initiateSession()
 			if err != nil {
+				resp.Body.Close()
 				return nil, err
 			}
+			if avisess.ctrlStatusCheckDisabled {
```

Comment:
Is the intention of not putting a `resp.Body.Close()` in this check is to check the response's body in the AKO's rest layer caller function?

Reply:
The response has to be read again. This is usually done in the function fetchBody(), which calls resp.Body.Close(), and fails if resp.Body is already closed.

Reply:
Got it. Thanks!